### DISCUSSION TITLE
Solution Explorer Context Menu

### DISCUSSION THREAD

**Discussion by dcastenholz** [msawczyn/EFDesigner#265] [Ideas]
Just found this area.  Maybe we can discuss here.

**Comment by dcastenholz** [comment]
Michael,

I have added a context menu to my own VSIX package to allow code generation by right-clicking on a .efmodel file and selecting "Generate Code".

My first question is whether you would consider me moving it to a pull request for inclusion in the designer.  The solution explorer context menu potentially has to be loaded prior to the loading of the designer menu system, but I can handle that in the code.  In the process of writing the menu code, I noticed that EFModelPackage uses ModelingPackage as its base class, rather than AsyncModelingPackage.  This complicates the code, and MS is actively trying to get developers to go async on all packages.  Would you consider switching to AsyncModelingPackage?  If you like the idea of the solution explorer context menu, I will do a PR, but if you would consider switching to AsyncModelingPackage, I will wait, as it will be a less invasive code change.

While we are at it, I also have added a context menu item that works on the designer surface to add an entity. Are you interested in a PR for that? I find it very convenient, but it is a slight deviation from the toolbox only philosophy.

Daniel

**Comment by msawczyn** [comment]
First, I always absolutely love it when folks submit pull requests! Please feel free to do that. For both suggestions .. they sound great!

Regarding the switch to `AsyncModelingPackage`, the designer is based on the Microsoft Modeling SDK (formerly known as DSL Tools). I'll have to dig in to see if that change is possible (might be easy, might not ... dunno) and what impact it has on the extension. But you're right, that is something that should be done with the big push towards asynchronous package loads. You'd think the Modeling SDK would have gone there already, but 🤷 .

**Reply by dcastenholz** [reply]
I created a new DSL not too long ago and it is async by default.  I don't remember even seeing the sync version as a choice.  I don't believe switching is a big deal, but as always, the devil is in the details.

I will do a pull request for each one.
